src/docfx/restore/Restore.cs

```diff
@@ -2,19 +2,84 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Diagnostics;
+using System.IO;
 using System.Threading.Tasks;
 
 namespace Microsoft.Docs.Build
 {
     internal static class Restore
     {
+        private static readonly string s_restoreDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".docfx", "git");
+
         public static Task Run(string docsetPath, CommandLineOptions options, ILog log)
         {
             // Restore has to use Config directly, it cannot depend on Docset,
             // because Docset assumes the repo to physically exist on disk.
             var config = Config.Load(docsetPath, options);
 
-            throw new NotImplementedException();
+            return ParallelUtility.ForEach(config.Dependencies.Values, (href, restoreChild) => RestoreDependentRepo(href, options, restoreChild));
+        }
+
+        /// <summary>
+        /// Get git repo information from git remote href, like https://github.com/org/repo#master
+        /// </summary>
+        /// <param name="remote">The git remote href</param>
+        /// <returns>The git repo information including local dir, git remote url and branch</returns>
+        public static (string dir, string url, string rev) GetGitInfo(string remote)
```

Comment:
Taking branch_mapping into consideration, I don't think string can match our request.

Reply:
will consider it later, current we support most simple case

---
In reply to: [182665813](https://github.com/dotnet/docfx/pull/2651#discussion_r182665813) [](ancestors = 182665813)